integrations/stripe/subscriptions/server/node/src/email.ts

```diff
@@ -0,0 +1,317 @@
+        organisation_name: 'your-organisation-name-here',
+    }
+
+    static async sendInvoicePaymentSucceededCard(
+        email: string,
+        data: {
+            invoice: Stripe.Invoice
+            paymentMethod: Stripe.PaymentMethod
+        },
+    ) {
+        const { invoice, paymentMethod } = data
+        this.sendEmail(
+            {
+                name: Template.InvoicePaymentSucceeded,
+                data: {
+                    invoice_id: invoice.id,
+                    invoice_url: invoice.hosted_invoice_url ?? '',
+                    amount: invoice.amount_paid?.toString() ?? '',
+                    currency: invoice.currency,
+                    payment_description: invoice.description ?? undefined,
+                    payment_method: `Card ${paymentMethod.card?.brand} ending in ${paymentMethod.card?.last4}`,
+                    receipt_date: formatDistanceToNow(invoice.created * 1000),
+                    receipt_id: invoice.receipt_number ?? '',
+                },
+            },
+            email,
+        )
+    }
+
+    static async sendInvoicePaymentFailedCard(
+        email: string,
+        data: {
+            invoice: Stripe.Invoice
+        },
+    ) {
+        const { invoice } = data
+        const nextAttempt =
+            invoice.next_payment_attempt !== null
+                ? formatDistanceToNow(invoice.next_payment_attempt * 1000)
+                : undefined
+        this.sendEmail(
+            {
+                name: Template.InvoicePaymentFailedCard,
+                data: {
+                    next_attempt_at: nextAttempt,
+                },
+            },
+            email,
+        )
+    }
+
+    static async sendSubscriptionCanceledPaymentFailed(email: string) {
+        this.sendEmail(
+            {
+                name: Template.SubscriptionCanceledPaymentFailed,
+            },
+            email,
+        )
+    }
+
+    static async sendSubscriptionCanceledUser(email: string) {
+        this.sendEmail(
+            {
+                name: Template.SubscriptionCanceledUser,
+            },
+            email,
+        )
+    }
+
+    // Show a short version of the address in the email
+    // E.g. 0x1234...7890
+    private static shortAddress(address: string) {
+        return (
+            address.substring(0, 6) +
+            '...' +
+            address.substring(address.length - 4)
+        )
+    }
+
+    static async sendInvoicePaymentSuccessWallet(
+        email: string,
+        data: {
+            invoice: Stripe.Invoice
+            charge: Charge
+        },
+    ) {
+        const { invoice, charge } = data
+
+        // Confirmed Charges will always have a transaction
+        if (!charge.transaction) return
+
+        const address = charge.source_address
+        // Show a short version of the address in the email
+        // E.g. 0x1234...7890
+        const shortAddress = this.shortAddress(address)
```

Comment:
By the way why using a shortened address and not sending out the full address? Might be more usable for the customer to know the full address?